src/WebJobs.Script/Workers/FunctionInvocationDispatcherState.cs

```diff
@@ -11,15 +11,26 @@ public enum FunctionInvocationDispatcherState
         Default,
 
         /// <summary>
-        /// The FunctionDispatcherState is starting.
+        /// The FunctionDispatcher is starting.
         /// </summary>
         Initializing,
 
         /// <summary>
-        /// The FunctionDispatcherState has been fully initialized and can accept direct function
+        /// The FunctionDispatcher has been fully initialized and can accept direct function
         /// invocations. All functions have been indexed. Listeners may not yet
         /// be not yet running.
         /// </summary>
-        Initialized
+        Initialized,
+
+        /// <summary>
+        /// The FunctionDispatcher was previously "Initialized" but no longer has any initialized
+        /// worker processes to handle function invocations.
+        /// </summary>
+        WorkerProcessRestarting,
+
+        /// <summary>
+        /// The FunctionDispatcher is disposing
+        /// </summary>
+        Disposing
```

Comment:
Also, add states: `Disposed` and `WorkerProcessRestarted` for consistency

Reply:
"WorkerProcessRestarted" is the same as "Initialized" and I don't want to distinguish those. Added "Disposed"